### ISSUE TITLE
Binds always empty in HostConfig

### ISSUE THREAD

**Issue by jbonjean** [docker-java/docker-java#436]
Hi,
It seems that binds fields (from HostConfig) are not properly populated after a container inspection, it always returns an empty array. A quick look at the code shows that the field seems to be deliberately ignored (@JsonIgnore). Is there a particular reason for this ?
Thanks.

**Comment by marcuslinke** [comment]
Are you sure that `binds` fields occurs in JSON response? Could you post example code please?  What docker version do you use?

**Comment by KostyaSha** [comment]
Maybe on old version because it changed var name?

**Comment by jbonjean** [comment]
Here the information you requested:
```
$ docker --version
Docker version 1.8.3, build f4bf5c7
```
Snippet of code to test the inspection:
```
DockerClient dockerClient = DockerClientBuilder.getInstance("unix:///var/run/docker.sock").build();
CreateContainerResponse container = dockerClient.createContainerCmd("busybox").withBinds(new Bind("/tmp/test", new Volume("/tmp"))).exec();
InspectContainerResponse containerInfo = dockerClient.inspectContainerCmd(container.getId()).exec();
System.out.println(containerInfo.getHostConfig().getBinds().length);
```
This snippet prints 0 for the binds length, which is incorrect.

```
$ docker inspect bdb94d832899
[
{
 ...
    "HostConfig": {
        "Binds": [
            "/tmp/test:/tmp:rw"
        ],
        "ContainerIDFile": "",
        "LxcConf": null,
        "Memory": 0,
 ...
}
]
```
I can also get the binds list with a manual call to the API:
```
curl -s 127.0.0.1:2375/containers/bdb94d832899/json
```

**Comment by marcuslinke** [comment]
@jbonjean The `@JsonIgnore` here https://github.com/docker-java/docker-java/blob/master/src/main/java/com/github/dockerjava/api/model/HostConfig.java#L101 should prevent the serialization of `Bind[]` because there is a special serializer for `Binds` objects (see https://github.com/docker-java/docker-java/blob/master/src/main/java/com/github/dockerjava/api/model/Binds.java#L34-L67). Maybe this prevents the deserialization of `Binds` field in general. Will try to reproduce ASAP.

**Comment by marcuslinke** [comment]
@jbonjean Just pushed d3c7df8df5b4d9565299445257526a4ade0ae610 which adds an assertion for binds in the appropriate test case. It passes with current 3.0.0-SNAPSHOT against docker v1.9.1 so the problem seems not to be at docker-java side. You could run your code with debug logging to see if binds are present in the JSON response coming from the daemon.

**Comment by jbonjean** [comment]
@marcuslinke I can confirm that it's working on the master, but not on the branch `2.x`. I did a git bisect and it seem that the bug had been fixed by 61bcaa70a32a5b5e3b6556dfb2b84e5cba575e8e.  Could this be ported to the branch 2.x?

**Comment by jbonjean** [comment]
@marcuslinke I backported only the changes from `src/main/java/com/github/dockerjava/api/model/Binds.java` into the `2.x` branch and  `binds` is now properly populated.

**Comment by marcuslinke** [comment]
@jbonjean So the problem is that binds format has changed sometimes and we need to know which docker version introduced the new serialization format. In the 2.x branch we need to stay compatible with docker 1.7 / v1.19, so someone need to evaluate this change against this specific docker version before I can backport it as currently there is no concept of multi API version support (at least for the serialization formats).

**Comment by KostyaSha** [comment]
@jbonjean could you post into gist.github.com full content of json dump?

**Comment by jbonjean** [comment]
@KostyaSha https://gist.github.com/jbonjean/9113ec342728cea208b7

**Comment by jbonjean** [comment]
@marcuslinke OK I understand. They don't make it easy to maintain a library :/

**Comment by marcuslinke** [comment]
@jbonjean So I fear you have to wait until 3.0.0 is released...

**Comment by jbonjean** [comment]
@marcuslinke OK, will Docker 1.8 be supported by the 3.0.0?

**Comment by marcuslinke** [comment]
@jbonjean It will support docker v1.9 / remote API v1.21 and should be mostly backward compatible. We try to be compatible but sometimes (as in this case) it is not possible atm.

**Comment by marcuslinke** [comment]
OK, closing this as fixed in current 3.0.0-SNAPSHOT.
